### PR TITLE
Interaction workflow

### PR DESCRIPTION
**Pull request by igorkorsukov**
Added description: 

* [AppShell](https://github.com/igorkorsukov/MuseScoreDocumentation/blob/develop_flux/Development/Architecture/AppShell.md) 
* [Interaction workflow](https://github.com/igorkorsukov/MuseScoreDocumentation/blob/develop_flux/Development/Architecture/InteractWorkflow.md)
* [Actions and ActionsDispatcher](https://github.com/igorkorsukov/MuseScoreDocumentation/blob/develop_flux/Development/Architecture/Actions.md)
* [Channels and Notifications](https://github.com/igorkorsukov/MuseScoreDocumentation/blob/develop_flux/Development/Architecture/ChannelsAndNotifications.md)

### CODE REVIEW

File: Development/Architecture/ChannelsAndNotifications.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+    s.saveFromChannel(p.channel());
+
+    p.process();
+}
+```
+
+## Notifications
+
+Notification designed to send notifications.
```
Comment (Marr11317) [inline Development/Architecture/ChannelsAndNotifications.md:198]: *are* designed?

File: Development/Architecture/InteractWorkflow.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+There is a list of `shortcuts`, `actions` are created from them (`QActions`), the action contains a command, when the action is triggered, the command pass to the main window (`MuseScore`), some commands are processed here, then pass to the current view (`ScoreView`), some commands are processed into the view, then pass to the model (`libmscore`), and there the rest of the teams are being processed.
+
+![mue_interaction_workflow_mu3.png](mue_interaction_workflow_mu3.png)
+
+In MuseScore 4, the main window ([AppShell](AppShell.md)) defines only the UI structure of the application and does not contain any action handlers.
+Another feature, in MuseSCore 4 there may be a model object (notation/libmscore), but the notation (score) view may not be loaded, for example, if we are working in the sequencer mode now, so the notation (score) cannot process actions, or several handlers of the same action may appear in different places, which leads to complexity and code duplication.
+For these reasons, in MuseScore 4, another workflow interaction
+
+![mue_interaction_workflow_mu4.png](mue_interaction_workflow_mu4.png)
```
Comment (Marr11317) [inline Development/Architecture/InteractWorkflow.md:12]: link not working (same above)

File: Development/Architecture/ChannelsAndNotifications.md (hunk 1 above)
Comment (igorkorsukov) [reply Development/Architecture/ChannelsAndNotifications.md]: Unfortunately I do not know much English.
To check the grammar, I use https://www.grammarly.com

Grammarly suggests that either nothing is needed, or `is` is needed (not are)

File: Development/Architecture/InteractWorkflow.md (hunk 2 above)
Comment (igorkorsukov) [reply Development/Architecture/InteractWorkflow.md:12]: working, this is the image, it shows 
https://github.com/igorkorsukov/MuseScoreDocumentation/blob/develop_flux/Development/Architecture/InteractWorkflow.md

**Pull request merged**